### PR TITLE
feat: add default video component, add custom controls

### PR DESCRIPTION
**Pull request by Kyzyl-ool**

### CODE REVIEW

**Review by yuberdysheva — DISMISSED** [review]

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

File: src/components/DefaultVideo/DefaultVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+export const DefaultVideo = React.forwardRef<HTMLVideoElement, DefaultVideoProps>((props, ref) => {
+    const {video, qa, customBarControlsClassName} = props;
+    const {controls, customControlsOptions, muted: initiallyMuted} = video;
+    const {muteButtonShown, positioning, type: customControlsType} = customControlsOptions || {};
+    const [isPaused, setIsPaused] = useState(false);
+    const [isMuted, setIsMuted] = useState(initiallyMuted);
+
+    const onPlayToggle = useCallback(() => {
+        setIsPaused((value) => {
+            if (typeof ref !== 'function') {
```
[the remark this answers is not in the document]
Comment (Kyzyl-ool) [reply src/components/DefaultVideo/DefaultVideo.tsx]: used another solution with useImperativeHandle

File: src/components/DefaultVideo/DefaultVideo.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <video
+                disablePictureInPicture
+                playsInline
+                // @ts-ignore
+                // eslint-disable-next-line react/no-unknown-property
+                pip="false"
+                className={b()}
+                ref={ref}
+                preload="metadata"
+                // @ts-ignore
```
[the remark this answers is not in the document]
Comment (Kyzyl-ool) [reply src/components/DefaultVideo/DefaultVideo.tsx]: not needed, removed the line

File: src/components/DefaultVideo/DefaultVideo.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        const {
+            muteButtonShown,
+            positioning,
+            type: customControlsType,
+        } = customControlsOptions || {};
+        const [isPaused, setIsPaused] = useState(false);
+        const [isMuted, setIsMuted] = useState(initiallyMuted);
+        const videoRef = useRef<HTMLVideoElement>(null);
+
+        useImperativeHandle(
```
Comment (gorgeousvlad) [inline src/components/DefaultVideo/DefaultVideo.tsx:34]: what is the point of `useImperativeHandle` usage here?

Comment (Kyzyl-ool) [reply src/components/DefaultVideo/DefaultVideo.tsx:40]: added comment

**Review by gorgeousvlad — APPROVED** [review]

**Pull request merged**